crates/bevy_scene/src/serde.rs

```diff
@@ -33,7 +34,14 @@ impl<'a> Serialize for SceneSerializer<'a> {
     where
         S: serde::Serializer,
     {
-        let mut state = serializer.serialize_struct(SCENE_STRUCT, 1)?;
+        let mut state = serializer.serialize_struct(SCENE_STRUCT, 2)?;
+        state.serialize_field(
+            SCENE_RESOURCES,
+            &ComponentsSerializer {
```

Comment:
Should we consider renaming this struct to be more generic? Something that works for components, resources, and any other list of `Box<dyn Reflect>`?

Reply:
I went with ReflectArraySerializer, but I admit I'm not good with names. If anyone has a better idea, I'm all ears :)